src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -2895,6 +2897,451 @@ public void Inheritance_17()
+    IL_0008:  ldarg.2
+    IL_0009:  ldarg.0
+    IL_000a:  call       ""int B.Y.get""
+    IL_000f:  stind.i4
+    IL_0010:  ret
+}");
+
+            var deconstruct = ((CSharpCompilation)verifier.Compilation).GetMember<MethodSymbol>("B.Deconstruct");
+            Assert.Equal(2, deconstruct.ParameterCount);
+
+            Assert.Equal(RefKind.Out, deconstruct.Parameters[0].RefKind);
+            Assert.Equal("X", deconstruct.Parameters[0].Name);
+
+            Assert.Equal(RefKind.Out, deconstruct.Parameters[1].RefKind);
+            Assert.Equal("Y", deconstruct.Parameters[1].Name);
+
+            Assert.True(deconstruct.ReturnsVoid);
+            Assert.False(deconstruct.IsVirtual);
+            Assert.False(deconstruct.IsStatic);
+            Assert.Equal(Accessibility.Public, deconstruct.DeclaredAccessibility);
+        }
+
+        [Fact]
+        public void Deconstruct_Nested()
+        {
+            var source =
+@"using System;
+
+data class B(int X, int Y);
+
+data class C(B B, int Z)
+{
+    public static void Main()
+    {
+        M(new C(new B(1, 2), 3));
+    }
+
+    static void M(C c)
+    {
+        switch (c)
+        {
+            case C(B(int x, int y), int z):
+                Console.Write(x);
+                Console.Write(y);
+                Console.Write(z);
+                break;
+        }
+    }
+}
+";
+
+            var verifier = CompileAndVerify(source, expectedOutput: "123");
+            verifier.VerifyDiagnostics();
+
+            verifier.VerifyIL("B.Deconstruct", @"
+{
+    // Code size       17 (0x11)
+    .maxstack  2
+    IL_0000:  ldarg.1
+    IL_0001:  ldarg.0
+    IL_0002:  call       ""int B.X.get""
+    IL_0007:  stind.i4
+    IL_0008:  ldarg.2
+    IL_0009:  ldarg.0
+    IL_000a:  call       ""int B.Y.get""
+    IL_000f:  stind.i4
+    IL_0010:  ret
+}");
+
+            verifier.VerifyIL("C.Deconstruct", @"
+{
+    // Code size       17 (0x11)
+    .maxstack  2
+    IL_0000:  ldarg.1
+    IL_0001:  ldarg.0
+    IL_0002:  call       ""B C.B.get""
+    IL_0007:  stind.ref
+    IL_0008:  ldarg.2
+    IL_0009:  ldarg.0
+    IL_000a:  call       ""int C.Z.get""
+    IL_000f:  stind.i4
+    IL_0010:  ret
+}");
+        }
+
+        [Fact]
+        public void Deconstruct_PropertyCollision()
+        {
+            var source =
+@"using System;
+
+data class B(int X, int Y)
+{
+    public int X => 3;
+
+    static void M(B b)
+    {
+        switch (b)
+        {
+            case B(int x, int y):
+                Console.Write(x);
+                Console.Write(y);
+                break;
+        }
+    }
+
+    static void Main()
+    {
+        M(new B(1, 2));
+    }
+}
+";
+            var verifier = CompileAndVerify(source, expectedOutput: "32");
+            verifier.VerifyDiagnostics();
+
+            var bClass = ((CSharpCompilation)verifier.Compilation).GetMember<NamedTypeSymbol>("B");
+            var deconstruct = bClass.GetMember("Deconstruct");
+            Assert.NotNull(deconstruct);
```

Comment:
Consider verifying the signature instead, here and in tests below.
```
Assert.Equal("void Deconstruct(out int X, out int Y)",
    comp.GetMember("B.Deconstruct").ToTestDisplayString());
```

Reply:
Done